hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/integration/ClientCompactSplitBrainTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.serialization.impl.compact.integration;
+
+import com.hazelcast.client.test.ClientTestSupport;
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.cluster.Member;
+import com.hazelcast.config.Config;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.internal.serialization.impl.compact.CompactTestUtil;
+import com.hazelcast.map.IMap;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import example.serialization.EmployeeDTO;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.Collections;
+import java.util.Set;
+
+import static com.hazelcast.test.SplitBrainTestSupport.blockCommunicationBetween;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class ClientCompactSplitBrainTest extends ClientTestSupport {
+
+    private final TestHazelcastFactory factory = new TestHazelcastFactory();
+
+    @After
+    public void tearDown() throws Exception {
+        factory.terminateAll();
+    }
+
+    @Test
+    public void testWithSplitBrain_clientReconnectsToOtherHalf() {
```

Comment:
What about changing the test name to something like `testLocalSchemasAreSentWithSplitBrain_clientReconnectsToOtherHalf` ? We can remove WithSplitBrain part as well because the test suite is about split brain, resulting in `testLocalSchemasAreSentWhenClientReconnectsToOtherHalf`.